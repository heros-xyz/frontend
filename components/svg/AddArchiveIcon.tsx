import { Icon, IconProps } from "@chakra-ui/react";
const AddArchiveIcon: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 32 32"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M31 16C31 24.2843 24.2843 31 16 31C7.71573 31 1 24.2843 1 16C1 7.71573 7.71573 1 16 1C24.2843 1 31 7.71573 31 16Z" />
      <path
        d="M16 11V16M16 16V21M16 16H21M16 16H11M31 16C31 24.2843 24.2843 31 16 31C7.71573 31 1 24.2843 1 16C1 7.71573 7.71573 1 16 1C24.2843 1 31 7.71573 31 16Z"
        stroke="currentColor"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Icon>
  );
};
export default AddArchiveIcon;
