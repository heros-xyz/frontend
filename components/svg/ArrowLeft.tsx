import { Icon, IconProps } from "@chakra-ui/react";

export const ArrowLeft: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 14 14"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.25 7L0.75 7M0.75 7L6.375 1.375M0.75 7L6.375 12.625"
        stroke="white"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Icon>
  );
};
