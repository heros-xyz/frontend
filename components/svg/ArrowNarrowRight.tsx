import { Icon, IconProps } from "@chakra-ui/react";

export const ArrowNarrowRight: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 18 14"
      fill="currentColor"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.1667 8.33317L16.5 4.99984M16.5 4.99984L13.1667 1.6665M16.5 4.99984L1.5 4.99984"
        stroke="currentColor"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Icon>
  );
};
