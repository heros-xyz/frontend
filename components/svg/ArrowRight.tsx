import { Icon, IconProps } from "@chakra-ui/react";

export const ArrowRight: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 15 14"
      fill="currentColor"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1.25 7L13.75 7M13.75 7L8.125 1.375M13.75 7L8.125 12.625"
        stroke="#33EFEF"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Icon>
  );
};
