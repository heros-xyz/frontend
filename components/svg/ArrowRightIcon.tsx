import { Icon, IconProps } from "@chakra-ui/react";

export const ArrowRightIcon: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 18 16"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1.5 8L16.5 8M16.5 8L9.75 1.25M16.5 8L9.75 14.75"
        stroke="currentColor"
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Icon>
  );
};
