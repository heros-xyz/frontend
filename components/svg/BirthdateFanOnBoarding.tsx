import { Icon, IconProps } from "@chakra-ui/react";

export const BirthdateFanOnBoarding: React.FC<IconProps> = (props) => {
  return (
    <Icon
      {...props}
      viewBox="0 0 151 150"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle cx="75.5" cy="75" r="74.5" fill="white" stroke="#DDDDDD" />
      <g clipPath="url(#clip0_1512_24344)">
        <path
          d="M108.364 67.7274C106.455 64.0001 61.1826 65.3638 44.9099 72.5456C39.0008 75.182 42.4553 113.455 52.4553 114.637C60.2735 115.546 77.0917 114.364 84.0917 113C86.9099 114.909 96.0917 111.091 96.0008 106.546C95.9099 104 94.4553 101.909 93.0917 100.546C94.4553 99.3638 95.4553 98.0001 95.7281 96.5456C96.1826 93.6365 94.0917 91.2729 92.3644 89.9092C94.4553 88.6365 96.0917 86.8183 96.4553 84.9092C96.9099 82.2729 94.6371 79.4547 92.5462 77.5456C105.637 75.9092 110.455 71.6365 108.364 67.7274Z"
          fill="#FAC484"
        />
        <path
          d="M94.0012 109.545C94.0012 109.545 85.4557 115.818 83.4557 112.273C81.0921 108.091 85.3648 105.273 85.3648 105.273C85.3648 105.273 79.3648 107.364 78.0921 103.727C76.8193 100.091 77.4557 97.3636 83.8193 94.6363C83.8193 94.6363 76.0012 97.3636 74.4557 93.4545C72.3648 88.0909 78.6375 83.6363 85.4557 80.2727"
          stroke="black"
          strokeWidth="1.5"
          strokeMiterlimit="10"
        />
        <path
          d="M68.0004 69.5454C73.0004 66.909 76.7276 72.8181 72.4549 77.2727C68.0913 81.6363 60.7276 82.8181 60.7276 82.8181C60.7276 82.8181 72.4549 91.9999 60.6367 95.6363"
          stroke="black"
          strokeWidth="1.5"
          strokeMiterlimit="10"
        />
        <path
          d="M89.0921 67.2728C88.5466 67.2728 88.183 66.9091 88.0921 66.3637C88.0012 64.1818 88.4557 60.6364 88.4557 60.5455C88.5466 60 89.0012 59.6364 89.5466 59.7273C90.0921 59.8182 90.4557 60.2728 90.3648 60.8182C90.3648 60.8182 89.9103 64.2728 90.0012 66.2728C90.0921 66.7273 89.7285 67.1818 89.0921 67.2728Z"
          fill="#F37421"
        />
        <path
          d="M83.1819 67.4544C82.6365 67.4544 82.2729 67.0908 82.1819 66.5453C82.091 64.3635 82.5456 60.818 82.5456 60.7271C82.6365 60.1817 83.091 59.818 83.6365 59.9089C84.1819 59.9999 84.5456 60.4544 84.4547 60.9999C84.4547 60.9999 84.0001 64.4544 84.091 66.4544C84.1819 66.9999 83.8183 67.4544 83.1819 67.4544C83.2729 67.4544 83.1819 67.4544 83.1819 67.4544Z"
          fill="#F37421"
        />
        <path
          d="M93.7275 59.7273C95.7275 57.8182 103.546 49.8182 99.3639 50.2728C98.7275 50.3637 97.273 51.3637 97.273 51.3637C97.273 51.3637 98.6366 48.7273 97.273 48.1818C96.0003 47.6364 93.8184 49.9091 93.8184 49.8182C93.7275 49.6364 95.3639 47.8182 94.5457 47.0909C94.1821 46.7273 92.7275 46.7273 90.5457 49.2728C89.7275 50.2728 92.9093 40.4546 91.3639 37.6364C89.4548 34.1818 82.3639 33.3637 79.273 39.9091C76.1821 46.4546 73.4548 57.4546 81.7275 61.0909C87.6366 63.6364 91.7275 61.7273 93.7275 59.7273Z"
          fill="#FEBF2E"
        />
        <path
          d="M80.3638 40.0908L74.9093 41.4545H74.8184L78.7275 44.9999L80.3638 40.0908Z"
          fill="#F37421"
        />
      </g>
      <defs>
        <clipPath id="clip0_1512_24344">
          <rect
            width="66.8182"
            height="80"
            fill="white"
            transform="translate(42 35)"
          />
        </clipPath>
      </defs>
    </Icon>
  );
};
