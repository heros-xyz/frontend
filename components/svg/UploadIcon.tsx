export const UploadIcon = () => {
  return (
    <svg
      width="21"
      height="20"
      viewBox="0 0 21 20"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.83398 13.3332L3.83398 14.1665C3.83398 15.5472 4.95327 16.6665 6.33398 16.6665L14.6673 16.6665C16.048 16.6665 17.1673 15.5472 17.1673 14.1665L17.1673 13.3332M13.834 6.6665L10.5006 3.33317M10.5006 3.33317L7.16732 6.6665M10.5006 3.33317L10.5006 13.3332"
        stroke="#D2FFFA"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
};
