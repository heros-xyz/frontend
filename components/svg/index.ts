export * from "./Checked";
export * from "./ChervonDown";
export * from "./Menu";
export * from "./Alert";
export * from "./ArrowNarrowRight";
export * from "./ArrowRight";
export * from "./BirthdateFanOnBoarding";
export * from "./PageInformation";
