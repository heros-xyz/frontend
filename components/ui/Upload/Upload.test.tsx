import { render } from "@testing-library/react";
import { UploadIcon } from "@/components/svg/UploadIcon";
import FileUpload from ".";

describe("FileUpload", () => {
  it("renders file upload", () => {
    const onChange = jest.fn();
    const { container } = render(
      <FileUpload
        text={"upload image"}
        icon={<UploadIcon />}
        onChange={onChange}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
