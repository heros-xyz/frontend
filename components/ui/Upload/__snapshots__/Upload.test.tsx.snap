// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload renders file upload 1`] = `
<div>
  <label
    class="chakra-button css-t7n79d"
    data-testid="file-upload"
  >
    upload image
    <input
      accept="image/jpeg,image/png"
      data-testid="file-input"
      hidden=""
      type="file"
    />
    <span
      class="chakra-button__icon css-1hzyiq5"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 21 20"
        width="21"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.83398 13.3332L3.83398 14.1665C3.83398 15.5472 4.95327 16.6665 6.33398 16.6665L14.6673 16.6665C16.048 16.6665 17.1673 15.5472 17.1673 14.1665L17.1673 13.3332M13.834 6.6665L10.5006 3.33317M10.5006 3.33317L7.16732 6.6665M10.5006 3.33317L10.5006 13.3332"
          stroke="#D2FFFA"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </label>
</div>
`;
